aws/cloudformation/iam.yml.erb

```diff
@@ -835,6 +835,90 @@ roles.each do |name, config| %>
+  DeployMarketingSitesPolicy<%=environment_type%>:
+    Type: AWS::IAM::Policy
+    Version: 2012-10-17
+    Properties:
+      PolicyName: DeployMarketingSitesPolicy<%=environment_type%>
+      Users:
+        - !Ref GithubDeployerMarketingSites<%=environment_type%>
+      PolicyDocument:
+        Statement:
+          - Effect: Allow
+            Action:
+              - cloudformation:CreateStack
+              - cloudformation:DescribeStacks
+              - cloudformation:CreateChangeSet
+              - cloudformation:DescribeChangeSet
+              - cloudformation:DeleteChangeSet
+              - cloudformation:ExecuteChangeSet
+            # The Stack Name must match this naming prefix convention. Examples:
+            # marketing-sites-production-code-org
+            # marketing-sites-production-hourofcode-com
+            # marketing-sites-test-teachai-org
+            # marketing-sites-test-code-org
+            # marketing-sites-development-code-org-my-feature-branch
+            # marketing-sites-development-code-org-my-other-branch
+            Resource: !Sub "arn:aws:cloudformation:${AWS::Region}:${AWS::AccountId}:stack/marketing-sites-<%=environment_type%>-*"
+          # Deployment User can Get and Pass any Fargate Task and Task Execution Roles for the current environment type.
+          - Effect: Allow
+            Action:
+              - iam:GetRole
+              - iam:PassRole
+            Resource: marketing-sites-<%=environment_type%>-*"
+          - Effect: Allow
+            Action:
+              - ecs:RegisterTaskDefinition
+            Resource: !Sub "arn:aws:ecs:${AWS::Region}:${AWS::AccountId}:task-definition/marketing-sites-<%=environment_type%>-*:*"
+          - Effect: Allow
+            Action:
+              - ecs:DeregisterTaskDefinition
+              - ecs:DescribeTaskDefinition
+              - elasticloadbalancing:DescribeLoadBalancers
+              - ec2:DescribeSecurityGroups
+            Resource: '*'
+          - Effect: Allow
+            Action:
+              - ecs:DescribeServices
+              - ecs:UpdateService
+            Resource: !Sub "arn:aws:ecs:${AWS::Region}:${AWS::AccountId}:service/marketing-sites-<%=environment_type%>-*/marketing-sites-<%=environment_type%>-*"
+          - Effect: Allow
+            Action:
+              - s3:PutObject
+              - s3:ListBucket
+            Resource:
+              # For the marketing site `my-feature-branch-code.marketing-sites.dev-code.org` the NextJS assets Bucket
+              # will be named `org.dev-code.marketing-sites.my-feature-branch.static-assets` so the `development`
+              # Deployer should have permission to all Buckets with the naming pattern `org.dev-code*static-assets`
+              - "<%=configuration[:base_domain_name].split('.').tap(&:reverse!).join('.')%>*static-assets"
+              - "<%=configuration[:base_domain_name].split('.').tap(&:reverse!).join('.')%>*static-assets/*"
+<% end %>
```

Comment:
To note some more permissions to add for [Cloudfront invalidation](https://github.com/code-dot-org/code-dot-org/pull/65707) and [autoscaling features](https://github.com/code-dot-org/code-dot-org/pull/65698):

```
  - Effect: Allow
    Action:
      - 'iam:GetRole'
    Resource:  !GetAtt ECSAutoScalingRole.Arn
  - Effect: Allow
    Action:
      - "cloudfront:CreateInvalidation"
    Resource: !Sub 'arn:aws:cloudfront::${AWS::AccountId}:distribution/${CloudFrontDistribution}'
```